### ISSUE TITLE
Cant compile with 5.1 c++ project

### ISSUE THREAD

**Issue by pasunna** [neil3d/UAnimatedTexture5#9]
The plugin work ok when try open in blueprint project
but when add c++ it need to rebuild from source
So I try rebuild from source
by this method
https://www.youtube.com/watch?v=sC0gnfYzFzU
but it fail with this log
```
[120/120] Compile Module.AnimatedTexture.cpp
G:\Plugin_51\AnimatedTexturePluginNew\HostProject\Plugins\AnimatedTexturePlugin\Source\AnimatedTexture\Private\AnimatedTexture2D.cpp(132): warning C4996: 'UTexture::Resource': Use GetResource() / SetResource() accessors instead. Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
G:\Plugin_51\AnimatedTexturePluginNew\HostProject\Plugins\AnimatedTexturePlugin\Source\AnimatedTexture\Private\AnimatedTextureResource.cpp(77): warning C4996: 'RHICreateTexture2D': FRHITexture2D is deprecated, please use RHICreateTexture(const FRHITextureCreateDesc&). Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
Total time in Parallel executor: 30.63 seconds
Total execution time: 34.54 seconds
Took 34.6220711s to run dotnet.exe, ExitCode=0
ERROR: System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'startIndex')

```
the rest
 ```
         at System.Globalization.CompareInfo.IndexOf(String source, String value, Int32 startIndex, Int32 count, CompareOptions options)
          at System.String.IndexOf(String value, Int32 startIndex, Int32 count, StringComparison comparisonType)
          at EpicGames.Core.FileFilter.AddRule(String pattern, FileFilterType type) in D:\build\++UE5\Sync\Engine\Saved\CsTools\Engine\Source\Programs\Shared\EpicGames.Core\FileFilter.cs:line 322
          at EpicGames.Core.FileFilter.AddRuleForFile(FileReference file, DirectoryReference baseDirectory, FileFilterType type) in D:\build\++UE5\Sync\Engine\Saved\CsTools\Engine\Source\Programs\Shared\EpicGames.Core\FileFilter.cs:line 179
          at BuildPlugin.FilterPluginFiles(FileReference PluginFile, IEnumerable`1 BuildProducts, IEnumerable`1 TargetPlatforms) in D:\build\++UE5\Sync\Engine\Saved\CsTools\Engine\Source\Programs\AutomationTool\Scripts\BuildPluginCommand.Automation.cs:line 426
          at BuildPlugin.PackagePlugin(FileReference SourcePluginFile, IEnumerable`1 BuildProducts, DirectoryReference TargetDir, Boolean bUnversioned, IEnumerable`1 TargetPlatforms) in D:\build\++UE5\Sync\Engine\Saved\CsTools\Engine\Source\Programs\AutomationTool\Scripts\BuildPluginCommand.Automation.cs:line 390
          at BuildPlugin.ExecuteBuild() in D:\build\++UE5\Sync\Engine\Saved\CsTools\Engine\Source\Programs\AutomationTool\Scripts\BuildPluginCommand.Automation.cs:line 124
          at AutomationTool.BuildCommand.Execute() in D:\build\++UE5\Sync\Engine\Saved\CsTools\Engine\Source\Programs\AutomationTool\AutomationUtils\BuildCommand.cs:line 344
          at AutomationTool.BuildCommand.ExecuteAsync() in D:\build\++UE5\Sync\Engine\Saved\CsTools\Engine\Source\Programs\AutomationTool\AutomationUtils\BuildCommand.cs:line 353
          at AutomationTool.Automation.ExecuteAsync(List`1 CommandsToExecute, Dictionary`2 Commands) in D:\build\++UE5\Sync\Engine\Saved\CsTools\Engine\Source\Programs\AutomationTool\AutomationUtils\Automation.cs:line 257
          at AutomationTool.Automation.ProcessAsync(ParsedCommandLine AutomationToolCommandLine, StartupTraceListener StartupListener, HashSet`1 ScriptModuleAssemblies) in D:\build\++UE5\Sync\Engine\Saved\CsTools\Engine\Source\Programs\AutomationTool\AutomationUtils\Automation.cs:line 156
       (see C:\Users\pazhentai\AppData\Roaming\Unreal Engine\AutomationTool\Logs\C+UE_5.1\Log.txt for full exception trace)
AutomationTool executed for 0h 3m 50s
AutomationTool exiting with ExitCode=1 (Error_Unknown)
BUILD FAILED
```

**Comment by neil3d** [comment]
Hi, thanks for the report — sorry for the long silence on this one.

The `master` branch has now been brought up to date and officially supports **UE 5.3 ~ 5.7**. The issues you saw on 5.1 / 5.2 (deprecated `UTexture::Resource`, `Pure virtual not implemented (UTexture::GetTextureClass)`, etc.) are all resolved in the current code.

Please pull the latest `master` and try again:
- https://github.com/neil3d/UAnimatedTexture5

A few notes:
- UE 5.0 / 5.1 / 5.2 are no longer actively maintained. If you must stay on those versions, the precompiled builds shared by community members in #13 still work.
- If you hit a build failure on 5.3+, please open a new issue with the full log and your engine version.

Closing this for now — feel free to reopen if anything still breaks on a supported version.
